Scripts/Runtime/Jobs/ValueAccessController/NativeArrayAccessController.cs

```diff
@@ -0,0 +1,14 @@
+using Unity.Collections;
+using Unity.Entities;
+
+namespace Anvil.Unity.DOTS.Jobs
+{
+    /// <summary>
+    /// A wrapper for NativeArray that provides a way to manage async access to the array.
+    /// </summary>
+    public class NativeArrayAccessController<T> : AbstractValueAccessController<NativeArray<T>> where T : struct, IComponentData
```

Comment:
Does `T` need to be `IComponentData`? 

What if you want to but in `IBufferElement` or any other `struct` really? I think we just care about making sure it's safe to read/write to the `NativeArray` but what goes in there just needs to be a `struct`

Reply:
Yea I don't know why I had that. At one point it fixed an issue but doesn't seem to be required anymore. Removed.

Reply:
👍